server/routes/prisonerIncentiveLevelDetails.ts

```diff
@@ -65,109 +63,162 @@ const SYSTEM_USERS = ['INCENTIVES_API']
 const hmppsAuthClient = new HmppsAuthClient(
   new TokenStore(createRedisClient('routes/prisonerIncentiveLevelDetails.ts')),
 )
+
+const manageUsersApiClient = new ManageUsersApiClient()
+
+
 export default function routes(router: Router): Router {
   const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
 
   get('/:prisonerNumber', async (req, res) => {
-
+    const { prisonerNumber } = req.params
     const systemToken = await hmppsAuthClient.getSystemClientToken(res.locals.user.username)
     const prisonApi = new PrisonApi(systemToken);
-    const { prisonerNumber } = req.params
     const incentivesApi = new IncentivesApi(systemToken)
-    const offenderSearchClient = new OffenderSearchClient(systemToken)
-
-    const incentiveLevelDetails: IepSummaryForBookingWithDetails= await incentivesApi.getIncentiveSummaryForPrisoner(prisonerNumber)
-    const currentIncentiveLevel = incentiveLevelDetails.iepLevel
-
-    const nextReviewDate = moment(incentiveLevelDetails.nextReviewDate, 'YYYY-MM-DD HH:mm')
-    const reviewDaysOverdue = newDaysSince(nextReviewDate)
-    const { agencyId, incentiveLevel, fromDate, toDate } = req.query
-
-    // TODO: FIX TS-IGNORES
-    //@ts-ignore
-    const fromDateFormatted = moment(fromDate, 'DD/MM/YYYY')
-    // @ts-ignore
-    const toDateFormatted = moment(toDate, 'DD/MM/YYYY')
-
-    // Offenders are likely to have multiple IEPs at the same agency.
-    // By getting a unique list of users and agencies, we reduce the duplicate
-    // calls to the database.
-    const uniqueUserIds = Array.from(new Set(incentiveLevelDetails.iepDetails.map((details) => details.userId)))
-    const uniqueAgencyIds = Array.from(new Set(incentiveLevelDetails.iepDetails.map((details) => details.agencyId)))
-
-    // Only get users that map to a user in the prison staff table
-    const users = (
-      await Promise.all(
-        uniqueUserIds
-          .filter((userId) => Boolean(userId))
-          .map((userId: string) => {
-            if (SYSTEM_USERS.includes(userId)) {
-              return {
-                username: userId,
-                firstName: 'System',
-                lastName: '',
-              }
-            }
-            return prisonApi.getStaffDetails(systemToken, userId)
-          })
-      )
-    ).filter(notEmpty)
 
-    const establishments: any[] = await Promise.all(uniqueAgencyIds.filter((id) => Boolean(id)).map((id) => prisonApi.getAgencyDetails(systemToken, id))
-    )
-    const levels = Array.from(new Set(incentiveLevelDetails.iepDetails.map((details) => details.iepLevel))).sort()
+    try {
```

Comment:
This is a bit try block, what's the rational for this? Just redirect on any error? Is this what DPS was doing?